### ISSUE TITLE
Stutter lint should not warn about `parse::Parser`.

### ISSUE THREAD

**Issue by kennytm** [rust-lang-nursery/rust-clippy#2212]
Repro case:

```rust
#![warn(stutter)]
#![allow(dead_code)]
mod parse {
    // ok
    pub struct Parse;
    // warning: item name starts with its containing module's name
    pub struct Parser;
}
fn main() {}
```

The stutter lint should not warn when the module name isn't a full word in the structure (unless you think we should rename the struct to `parse::R` ಠ_ಠ)

```
$ cargo clippy --version
0.0.169
```

**Comment by oli-obk** [comment]
We can check whether the character after the idential part is an underscore or captialized and only lint in that case.

**Comment by kennytm** [comment]
If the identical part follows a number the lint should also be generated (e.g. `open::Open7Zip`).
